tests/fixtures/dummies.py

```diff
@@ -28,27 +27,22 @@
 class DummySecretsHandler(SecretsHandlerPort):
     """Dummy SecretsHandler implementation for testing.
 
-    It can be set to fail when `get_secrets` is called to mimic an error.
+    `secrets` is a dictionary that maps vault paths to lists of secrets.
     """
 
     def __init__(
-        self, secrets: list[str] | None = None, fail_on_get_secrets: bool = False
+        self,
+        secrets: dict[str, list[str]] | None = None,
     ):
-        self.secrets = secrets if secrets else []
-        self.fail_on_get_secrets = fail_on_get_secrets
-
-    def get_secrets(self) -> list[str]:
-        """Get all secrets currently stored.
+        self.secrets = secrets if secrets else {}
 
-        If `fail_on_get_secrets` is set, it will raise an `InvalidPath` error.
-        """
-        if self.fail_on_get_secrets:
-            raise InvalidPath("Testing failure")
-        return self.secrets
+    def get_secrets(self, vault_path: str) -> list[str]:
+        """Get all secrets currently stored for the specified vault."""
+        return self.secrets.get(vault_path, [])
 
-    def delete_secrets(self) -> None:
+    def delete_secrets(self, vault_path: str) -> None:
         """Delete all secrets stored in the vault."""
-        self.secrets = []
+        self.secrets.pop(vault_path, None)
```

Comment:
This would also remove the key from the dict.
Is this intended?

Reply:
I've removed the whole dummy class and replaced it with Mock in [536198b](https://github.com/ghga-de/state-management-service/pull/8/commits/536198b764f08d1500e106d940e449c3d4167389)